### ISSUE TITLE
Query methods should be able to return probability values of specific states.

### ISSUE THREAD

**Issue by ankurankan** [pgmpy/pgmpy#779]
Currently query methods can only return a Factor/TabularCPD instance over a variable. It should also work in the case of queries like:

```python
model.query([('var', 0)], evidence=[(something)]
```
In this case it should return the probability value of `P(var=0 | evidence)`.

**Comment by raghavg7796** [comment]
I am new here. Can I help to contribute ?

**Comment by yashu-seth** [comment]
@raghavg7796 Yes. Just let us know if you face any problem.

**Comment by ankurankan** [comment]
@raghavg7796 This should be fairly easy. Currently we return the whole `DiscreteFactor` object on calling `query`, you just need to modify it to return a specific value if the state is provided.

**Comment by raghavg7796** [comment]
Should we make a query method like value_query or something or it should be implicit in the query method only?

**Comment by raghavg7796** [comment]
And if implicit how we will distinguish between [A, B, C, D] and [A, 0, B, 2]

**Comment by ankurankan** [comment]
@raghavg7796 I am sorry the original comment was wrong. So, there should be two possible ways of querying.
```python
model_infer.query(['A', 'B'], evidence=evidence_values)
model_infer.query([('A', 0)], evidence=evidence_values)
```
In general the first argument would either be a 1-D array of 2-D array/list of tuples.

**Comment by raghavg7796** [comment]
Should value given for a variable be its one of the state or value should be between [0, cardinality]

**Comment by raghavg7796** [comment]
And I am getting another confusion that what if ('A', 0) is itself a variable. We can have tuples as variables also??

**Comment by yashu-seth** [comment]
@raghavg7796 It can be both. But for the time being you don't have to worry about this. Let us have a discussion on gitter and I'll explain it to you.

**Comment by ankurankan** [comment]
@yashu-seth @raghavg7796  I didn't think of the case when the variable names are tuples like in the case of Dynamic Bayesian Networks. Wouldn't there be an issue because of the new query argument ?

**Comment by yashu-seth** [comment]
@ankurankan Yes I was also thinking about this. There could be issues with the state name architecture too with tuple variable names.

**Comment by raghavg7796** [comment]
@ankurankan @yashu-seth I think we can use this with Bayesian Model only as inference with Bayesian Model requires string.

**Comment by raghavg7796** [comment]
There can be 3 inputs:
1. infer.query(variables = [('A', ('A', 0), ('A', 2), 'B'...
2. infer.query(variables = [('A', 0), 'B'... i,e, one variable and only one value
3.infer.query(variables = [('A', 0), ('B', 0)  ...i,e, only 1 type either all strings or all tuples
I have added support for Case 1 in my PR..please review it....

**Comment by ankurankan** [comment]
@yashu-seth @raghavg7796 I can't think of any clear solution to this. Any ideas ?

**Comment by raghavg7796** [comment]
@ankurankan @yashu-seth I have a idea. We can check for if a tuple is in list of variables. If it is we perform normal query, otherwise value query,

**Comment by ankurankan** [comment]
@raghavg7796 But we can differentiate between a value query and a variable which is a tuple.

**Comment by raghavg7796** [comment]
@ankurankan if a tuple is recieved like ('A', 0), we'll check in list of variables if it is there we give normal query, otherwise we'll assume user wants to ask P('A'=0)

**Comment by raghavg7796** [comment]
@ankurankan @yashu-seth I have considered the case of tuple variables. Please review.

**Comment by ankurankan** [comment]
@raghavg7796 I am not really sure if this is a good idea.

**Comment by raghavg7796** [comment]
@ankurankan But it is working fine. Can you please tell why or suggest any other change? Or if you want to hold it, lets hold it for a while till we have a better solution.

**Comment by ankurankan** [comment]
@raghavg7796 I would suggest to hold on till be find a good solution. Your solution for searching for the variables is kind of hackish and I don't like it.

**Comment by raghavg7796** [comment]
@ankurankan I think a separate method will also allow us to calculate probabilities like P(a = 1, b = 2|evidences) because otherwise as @ValerioBiscione mentioned there is no method to do thing like this.

**Comment by ValerioB88** [comment]
Guys, I think you must clearly say somewhere in the doc that query(variables=['a','b']) does not correspond to p(a,b), but returns p(a) and p(b). I think this behavior should be changed (not by adding a new method!), so that query(variables=['a','b']) returns p(a,b).

**Comment by ankurankan** [comment]
@raghavg7796 Both of these are two separate issues. Let's not mix them up. We can easily solve the issue with returning a joint factor. 

And I would not like to have a new method for computing values for specific states.
